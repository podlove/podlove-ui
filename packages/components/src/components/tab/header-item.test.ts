import { describe, test, expect } from 'vitest';
import { mount } from '@vue/test-utils'

import TabHeader from './HeaderItem.vue'

describe('TabHeaderItem', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(TabHeader)

      expect(wrapper.vm).toBeTruthy()
    })
  })

  describe('props', () => {
    test('should render without props', async () => {
      const wrapper = await mount(TabHeader, {
        slots: {
          icon: '<span>Icon</span>',
          title: '<span>title</span>'
        }
      })

      expect(wrapper.element).toMatchSnapshot()
    })

    const props = [
      {
        prop: 'index',
        value: 5
      },
      {
        prop: 'name',
        value: 'tab-name'
      },
      {
        prop: 'active',
        value: false
      },
      {
        prop: 'display',
        value: 'fixed'
      },
      {
        prop: 'color',
        value: '#fff'
      },
      {
        prop: 'background',
        value: '#fff'
      },
      {
        prop: 'colorActive',
        value: '#fff'
      },
      {
        prop: 'backgroundActive',
        value: '#fff'
      }
    ]

    props.forEach(({ prop, value }) => {
      test(`should render with prop ${prop}`, async () => {
        const wrapper = await mount(TabHeader, {
          slots: {
            icon: '<span>Icon</span>',
            title: '<span>title</span>'
          },
          props: {
            [prop]: value
          }
        })

        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
