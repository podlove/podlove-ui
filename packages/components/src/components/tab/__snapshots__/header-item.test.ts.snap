// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TabHeaderItem > props > should render with prop active 1`] = `
<li
  class="podlove-component-tab-header-item flex items-center justify-center w-full m-0 opacity-100 hover:opacity-75 h-8 native"
  data-v-78036178=""
  rel=""
  role="presentation"
>
  <button
    aria-controls=""
    aria-selected="false"
    class="flex items-center justify-center overflow-hidden pt-0 pb-0 pl-4 pr-4 align-middle text-center w-full h-full"
    data-test="podlove-component-tab-header-item-trigger-"
    data-v-78036178=""
    role="tab"
    tabindex="0"
  >
    <span
      aria-hidden="true"
      class="icon mr-1"
      data-v-78036178=""
    >
      
      <span>
        Icon
      </span>
      
    </span>
    <span
      class="uppercase ml-1"
      data-v-78036178=""
    >
      
      <span>
        title
      </span>
      
    </span>
    <!--v-if-->
  </button>
</li>
`;

exports[`TabHeaderItem > props > should render with prop background 1`] = `
<li
  background="#fff"
  class="podlove-component-tab-header-item flex items-center justify-center w-full m-0 opacity-100 hover:opacity-75 h-8 native"
  data-v-78036178=""
  rel=""
  role="presentation"
>
  <button
    aria-controls=""
    aria-selected="false"
    class="flex items-center justify-center overflow-hidden pt-0 pb-0 pl-4 pr-4 align-middle text-center w-full h-full"
    data-test="podlove-component-tab-header-item-trigger-"
    data-v-78036178=""
    role="tab"
    tabindex="0"
  >
    <span
      aria-hidden="true"
      class="icon mr-1"
      data-v-78036178=""
    >
      
      <span>
        Icon
      </span>
      
    </span>
    <span
      class="uppercase ml-1"
      data-v-78036178=""
    >
      
      <span>
        title
      </span>
      
    </span>
    <!--v-if-->
  </button>
</li>
`;

exports[`TabHeaderItem > props > should render with prop backgroundActive 1`] = `
<li
  backgroundactive="#fff"
  class="podlove-component-tab-header-item flex items-center justify-center w-full m-0 opacity-100 hover:opacity-75 h-8 native"
  data-v-78036178=""
  rel=""
  role="presentation"
>
  <button
    aria-controls=""
    aria-selected="false"
    class="flex items-center justify-center overflow-hidden pt-0 pb-0 pl-4 pr-4 align-middle text-center w-full h-full"
    data-test="podlove-component-tab-header-item-trigger-"
    data-v-78036178=""
    role="tab"
    tabindex="0"
  >
    <span
      aria-hidden="true"
      class="icon mr-1"
      data-v-78036178=""
    >
      
      <span>
        Icon
      </span>
      
    </span>
    <span
      class="uppercase ml-1"
      data-v-78036178=""
    >
      
      <span>
        title
      </span>
      
    </span>
    <!--v-if-->
  </button>
</li>
`;

exports[`TabHeaderItem > props > should render with prop color 1`] = `
<li
  class="podlove-component-tab-header-item flex items-center justify-center w-full m-0 opacity-100 hover:opacity-75 h-8 native"
  color="#fff"
  data-v-78036178=""
  rel=""
  role="presentation"
>
  <button
    aria-controls=""
    aria-selected="false"
    class="flex items-center justify-center overflow-hidden pt-0 pb-0 pl-4 pr-4 align-middle text-center w-full h-full"
    data-test="podlove-component-tab-header-item-trigger-"
    data-v-78036178=""
    role="tab"
    tabindex="0"
  >
    <span
      aria-hidden="true"
      class="icon mr-1"
      data-v-78036178=""
    >
      
      <span>
        Icon
      </span>
      
    </span>
    <span
      class="uppercase ml-1"
      data-v-78036178=""
    >
      
      <span>
        title
      </span>
      
    </span>
    <!--v-if-->
  </button>
</li>
`;

exports[`TabHeaderItem > props > should render with prop colorActive 1`] = `
<li
  class="podlove-component-tab-header-item flex items-center justify-center w-full m-0 opacity-100 hover:opacity-75 h-8 native"
  coloractive="#fff"
  data-v-78036178=""
  rel=""
  role="presentation"
>
  <button
    aria-controls=""
    aria-selected="false"
    class="flex items-center justify-center overflow-hidden pt-0 pb-0 pl-4 pr-4 align-middle text-center w-full h-full"
    data-test="podlove-component-tab-header-item-trigger-"
    data-v-78036178=""
    role="tab"
    tabindex="0"
  >
    <span
      aria-hidden="true"
      class="icon mr-1"
      data-v-78036178=""
    >
      
      <span>
        Icon
      </span>
      
    </span>
    <span
      class="uppercase ml-1"
      data-v-78036178=""
    >
      
      <span>
        title
      </span>
      
    </span>
    <!--v-if-->
  </button>
</li>
`;

exports[`TabHeaderItem > props > should render with prop display 1`] = `
<li
  class="podlove-component-tab-header-item flex items-center justify-center w-full m-0 opacity-100 hover:opacity-75 h-8 fixed"
  data-v-78036178=""
  rel=""
  role="presentation"
>
  <button
    aria-controls=""
    aria-selected="false"
    class="flex items-center justify-center overflow-hidden pt-0 pb-0 pl-4 pr-4 align-middle text-center w-full h-full"
    data-test="podlove-component-tab-header-item-trigger-"
    data-v-78036178=""
    role="tab"
    tabindex="0"
  >
    <span
      aria-hidden="true"
      class="icon mr-1"
      data-v-78036178=""
    >
      
      <span>
        Icon
      </span>
      
    </span>
    <span
      class="uppercase ml-1"
      data-v-78036178=""
    >
      
      <span>
        title
      </span>
      
    </span>
    <!--v-if-->
  </button>
</li>
`;

exports[`TabHeaderItem > props > should render with prop index 1`] = `
<li
  class="podlove-component-tab-header-item flex items-center justify-center w-full m-0 opacity-100 hover:opacity-75 h-8 native"
  data-v-78036178=""
  rel=""
  role="presentation"
>
  <button
    aria-controls=""
    aria-selected="false"
    class="flex items-center justify-center overflow-hidden pt-0 pb-0 pl-4 pr-4 align-middle text-center w-full h-full"
    data-test="podlove-component-tab-header-item-trigger-"
    data-v-78036178=""
    role="tab"
    tabindex="5"
  >
    <span
      aria-hidden="true"
      class="icon mr-1"
      data-v-78036178=""
    >
      
      <span>
        Icon
      </span>
      
    </span>
    <span
      class="uppercase ml-1"
      data-v-78036178=""
    >
      
      <span>
        title
      </span>
      
    </span>
    <!--v-if-->
  </button>
</li>
`;

exports[`TabHeaderItem > props > should render with prop name 1`] = `
<li
  class="podlove-component-tab-header-item flex items-center justify-center w-full m-0 opacity-100 hover:opacity-75 h-8 native"
  data-v-78036178=""
  rel="tab-name"
  role="presentation"
>
  <button
    aria-controls="tab-name"
    aria-selected="false"
    class="flex items-center justify-center overflow-hidden pt-0 pb-0 pl-4 pr-4 align-middle text-center w-full h-full"
    data-test="podlove-component-tab-header-item-trigger-tab-name"
    data-v-78036178=""
    role="tab"
    tabindex="0"
  >
    <span
      aria-hidden="true"
      class="icon mr-1"
      data-v-78036178=""
    >
      
      <span>
        Icon
      </span>
      
    </span>
    <span
      class="uppercase ml-1"
      data-v-78036178=""
    >
      
      <span>
        title
      </span>
      
    </span>
    <!--v-if-->
  </button>
</li>
`;

exports[`TabHeaderItem > props > should render without props 1`] = `
<li
  class="podlove-component-tab-header-item flex items-center justify-center w-full m-0 opacity-100 hover:opacity-75 h-8 native"
  data-v-78036178=""
  rel=""
  role="presentation"
>
  <button
    aria-controls=""
    aria-selected="false"
    class="flex items-center justify-center overflow-hidden pt-0 pb-0 pl-4 pr-4 align-middle text-center w-full h-full"
    data-test="podlove-component-tab-header-item-trigger-"
    data-v-78036178=""
    role="tab"
    tabindex="0"
  >
    <span
      aria-hidden="true"
      class="icon mr-1"
      data-v-78036178=""
    >
      
      <span>
        Icon
      </span>
      
    </span>
    <span
      class="uppercase ml-1"
      data-v-78036178=""
    >
      
      <span>
        title
      </span>
      
    </span>
    <!--v-if-->
  </button>
</li>
`;
